tests/test_requests.py

```diff
@@ -32,10 +33,29 @@
 from .compat import StringIO, u
 from .utils import override_environ
 
+from requests.packages.urllib3.exceptions import SNIMissingWarning
+
+# urllib3 sets this to only go off once, but we need it to
+# always fire for test_https_warnings to work
+warnings.simplefilter('always', SNIMissingWarning)
```

Comment:
Yeah, that's why I said it risks warning spam but didn't strictly say to remove it: I'm a bit aware that warning filters are awkwardly global. Is there nothing we can do about the spam at all?